### PR TITLE
Move ice nucleation J params to CLIMAParams

### PR DESCRIPTION
**Pull request by trontrytel**
This PR:
- moves ice nucleation parameters to CLIMAParameters
- specifies CLIMAParameters version in Compat

### CODE REVIEW

File: src/Common.jl [hunk 1]
```diff
@@ -124,24 +124,41 @@ function logistic_function_integral(x::FT, x_0::FT, k::FT) where {FT <: Real}
...
-function H2SO4_soln_saturation_vapor_pressure(x::FT, T::FT) where {FT <: Real}
+function H2SO4_soln_saturation_vapor_pressure(
+    prs::APS,
+    x::FT,
+    T::FT,
+) where {FT <: Real}
+
+    T_max::FT = CMP.H2SO4_sol_T_max(prs)
+    T_min::FT = CMP.H2SO4_sol_T_min(prs)
+    w_h::FT = CMP.H2SO4_sol_w_2(prs)
```
Comment (amylu00) [inline src/Common.jl:144]:
```suggestion
    w_2::FT = CMP.H2SO4_sol_w_2(prs)
```

File: src/Common.jl [hunk 2]
```diff
@@ -124,24 +124,41 @@ function logistic_function_integral(x::FT, x_0::FT, k::FT) where {FT <: Real}
...
+    c6::FT = CMP.H2SO4_sol_c6(prs)
+    c7::FT = CMP.H2SO4_sol_c7(prs)
 
-    @assert T < FT(235)
-    @assert T > FT(185)
+    @assert T < T_max
+    @assert T > T_min
 
-    w_h = 1.4408 * x
+    w_h = w_h * x
```
Comment (amylu00) [inline src/Common.jl:157]:
```suggestion
    w_h = w_2 * x
```

**Review by amylu00 — APPROVED** [review]
Suggesting a change in variable name for w_h so it's consistent with Luo's paper. Thanks Anna!

File: src/Common.jl (hunk 1 above)
Comment (trontrytel) [reply src/Common.jl:144]: done!

**Comment by trontrytel** [comment]
bors try

**Pull request merged**